use mamba2 for documentation build test in Circle CI (#3955)
ESMValGroup/ESMValTool: doc/sphinx/source/quickstart/installation.rst

    would like to keep.
    It is recommended that you create a new environment when updating ESMValTool.

.. note::
    There is also a pure-Python environment file ``esmvaltool_python.yml``
    which is a softlink of the ``environment_osx.yml`` file; this one is used

>     by any build that needs only Python packages (ie no NCL, R etc), currently

Reviewer: ```suggestion
    by any build that needs only Python packages (i.e. no NCL and R), currently
```

>     this is used by our documentation builds, but it could be used by any
>     other process needing just the Python dependencies.

Reviewer: ```suggestion
    this is used by our documentation builds, but it could be used by anyone
    needing just the Python dependencies.
```
